### PR TITLE
[ADD] supplier invoice number info, only in supplier tree and search …

### PR DESCRIPTION
**Pull request by archetipo**
…view
consider:

https://github.com/OCA/account-invoicing/pull/90#issuecomment-121628283
I've proposed a module , because is not a simple adding a field in a view, but implements a way to create and open your own inherited invoice view ,without modify the existing view, via the "mode=priority".

### CODE REVIEW

File: account_invoice_supplier_number_info/README.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+supplier_invoice_number field added in the tree and search view by
+inheritance
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/{project_repo}/issues/new?body=module:%20{module_name}%0Aversion:%20{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (eLBati) [inline account_invoice_supplier_number_info/README.rst]: please replace `{project_repo}` , `{module_name}` and `{version}`

Comment (archetipo) [inline account_invoice_supplier_number_info/README.rst]: ok tnx!

File: account_invoice_supplier_number_info/view/invoice_view.xml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        <record id="account.view_account_invoice_filter" model="ir.ui.view">
+            <field name="mode">primary</field>
+        </record>
+
+
+        <record model="ir.ui.view" id="view_account_invoice_supplier_tree">
+            <field name="name">view.account_invoice.supplier.tree</field>
+            <field name="model">account.invoice</field>
+            <field name="inherit_id" ref="account.invoice_tree" />
+            <field name="mode">primary</field>
```
Comment (pedrobaeza) [inline account_invoice_supplier_number_info/view/invoice_view.xml]: This line is not needed

File: account_invoice_supplier_number_info/view/invoice_view.xml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                    <field name="supplier_invoice_number"/>
+                </xpath>
+            </field>
+        </record>
+
+        <record id="account.action_invoice_tree2" model="ir.actions.act_window">
+            <field name="search_view_id" ref="view_account_invoice_supplier_search"/>
+        </record>
+
+        <record id="action_invoice_tree2_view1" model="ir.actions.act_window.view">
```
Comment (pedrobaeza) [inline account_invoice_supplier_number_info/view/invoice_view.xml:42]: Why do you need all of this?

**Comment by pedrobaeza** [comment]
I'm wondering if it isn't better to propose a patch on OCB/Odoo to include this field in the view

File: account_invoice_supplier_number_info/view/invoice_view.xml (hunk 3 above)
Comment (archetipo) [inline account_invoice_supplier_number_info/view/invoice_view.xml:42]: Hi @pedrobaeza 
primary function, needs to specific action for each view you want open directly.

Comment (pedrobaeza) [inline account_invoice_supplier_number_info/view/invoice_view.xml:42]: But I mean that I don't understand why you need to define a primary view where you can use the standard one

Comment (archetipo) [inline account_invoice_supplier_number_info/view/invoice_view.xml:42]: @pedrobaeza 
The purpose of this module is implement the field "supplier_invoice_number" ,and do this only in supplier invoice views(search , tree, form). to do this have used the primary mode in views records and handled with specific actions.
for more info: [Here](https://www.odoo.com/documentation/8.0/reference/views.html) section Inheritance -> View matching

Comment (pedrobaeza) [inline account_invoice_supplier_number_info/view/invoice_view.xml:42]: OK, sorry for the confusion, I thought there was a standard separated tree view for supplier invoices (as the form view one).

**Comment by pedrobaeza** [comment]
:+1:

File: account_invoice_supplier_number_info/README.rst [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+Invoice number:
+===============
+
+supplier_invoice_number field added in the tree and search view by
+inheritance
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
```
Comment (eLBati) [inline account_invoice_supplier_number_info/README.rst]: please replace `{project_repo}`

File: account_invoice_supplier_number_info/README.rst [hunk 5]
```diff
@@ -0,0 +1,39 @@
+ .. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Invoice number:
+===============
+
+supplier_invoice_number field added in the tree and search view by
+inheritance
+
```
Comment (eLBati) [inline account_invoice_supplier_number_info/README.rst:17]: You may want to add the 'try me on runbot' button https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst#usage

**Comment by eLBati** [comment]
some remarks, then :+1:

File: account_invoice_supplier_number_info/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    'author': "Agile Business Group,Odoo Community Association (OCA)",
+    'website': 'http://www.agilebg.com',
+    'license': 'AGPL-3',
+    "depends": [
+        'account'
+    ],
+    "data": [
+        'view/invoice_view.xml'
+    ],
+    "active": False,
```
Comment (tafaRU) [inline account_invoice_supplier_number_info/__openerp__.py]: This key is deprecated and it's an alias for auto_install whose default value is already False. So please remove it.

File: account_invoice_supplier_number_info/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': "Supplier Invoice Number Info",
+    'version': '0.1',
```
Comment (tafaRU) [inline account_invoice_supplier_number_info/__openerp__.py]: Please change it according to https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#version-numbers

Comment (tafaRU) [inline account_invoice_supplier_number_info/__openerp__.py]: @archetipo it seems to me that point hasn't been addressed properly. It misses Odoo major version at the beginning, isn'it?

**Comment by legalsylvain** [comment]
I don't understand why runbot is red.

**Comment by eLBati** [comment]
On 31 August 2015 at 12:38, Sylvain LE GAL notifications@github.com wrote:

> I don't understand why runbot is red.
> 
> Maybe because of

2015-08-25 12:16:01_WARNING_server
openerp.addons.account_asset_management_xls:30
https://github.com/OCA/account-invoicing/blob/e109bbd3884636952ab775757a5499956c661646/openerp/addons/account_asset_management_xls/__init__.py#L30
<module>
report_xls not available in addons path2015-08-25 12:18:44_INFO_runbot
odoo.runbot:0
https://github.com/OCA/account-invoicing/blob/e109bbd3884636952ab775757a5499956c661646/runbot#L0
test_all
Start test all modules2015-08-25 12:19:12_WARNING_server
openerp.addons.account_asset_management_xls:30
https://github.com/OCA/account-invoicing/blob/e109bbd3884636952ab775757a5499956c661646/openerp/addons/account_asset_management_xls/__init__.py#L30
<module>
report_xls not available in addons path
?

File: account_invoice_supplier_number_info/README.rst [hunk 8]
```diff
@@ -0,0 +1,46 @@
+ .. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
```
Comment (tafaRU) [inline account_invoice_supplier_number_info/README.rst:4]: Please modify the module title syntax as shown in https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: account_invoice_supplier_number_info/README.rst [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+Invoice number:
```
Comment (tafaRU) [inline account_invoice_supplier_number_info/README.rst]: Please remove the colon at the end.

File: account_invoice_supplier_number_info/README.rst [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+===============
+
+supplier_invoice_number field added in the tree and search view by
+inheritance
```
Comment (tafaRU) [inline account_invoice_supplier_number_info/README.rst]: I would improve a bit the module description.

File: account_invoice_supplier_number_info/__init__.py [hunk 11]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2011-2013 Agile Business Group (<http://www.agilebg.com>)
```
Comment (tafaRU) [inline account_invoice_supplier_number_info/__init__.py]: Please update copyright year.

File: account_invoice_supplier_number_info/__openerp__.py [hunk 12]
```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2011-2013 Agile Business Group (<http://www.agilebg.com>)
```
Comment (tafaRU) [inline account_invoice_supplier_number_info/__openerp__.py]: Same as above.

File: account_invoice_supplier_number_info/__openerp__.py [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': "Supplier Invoice Number Info",
+    'version': '1.1.0',
```
Comment (tafaRU) [inline account_invoice_supplier_number_info/__openerp__.py]: Please update it according to last OCA guidelines.

**Comment by archetipo** [comment]
@tafaRU thanks for review

**Comment by tafaRU** [comment]
@archetipo, please squash all commits in one, as this is a new module and there's no need for the commit history for the reviewed changes. When you do it, I'll merge, thanks!

**Comment by archetipo** [comment]
@tafaRU  thanks squash done

**Comment by tafaRU** [comment]
@archetipo, last thing: could you please modify the commit message?

Personally, I would use **Add  account_invoice_supplier_number_info module** to point out you're adding  something rather you are improving something.

Please ping me when you're done.

Thank you in advance.

**Comment by tafaRU** [comment]
Thanks @archetipo, I merge then.

**Pull request merged**